LaTeX/TeX.sublime-syntax

```diff
@@ -350,6 +247,122 @@ contexts:
     - include: brace-group-end
     - include: main
 
+###[ Registers ]################################################################
+
+  registers:
+    - match: (\\)({{registers}}){{endcs}}
+      captures:
+        1: punctuation.definition.backslash.tex
+        2: storage.type.tex
+      push: register-id
+
+  register-id:
+    - meta_scope: meta.register.tex
+    - match: \d+
+      scope: meta.number.integer.decimal constant.numeric.value.tex
+      set:
+        - tex-dimension-value
+        - tex-assignment
+    - match: (\\){{letter}}+
+      scope: support.function.general.tex
+      captures:
+        1: punctuation.definition.backslash.tex
+      set:
+        - tex-dimension-value
+        - tex-assignment
+    - include: paragraph-pop
+    - include: else-pop
+
+  register-allocations:
+    - match: (\\)new{{registers}}{{endcs}}
+      scope: keyword.declaration.register.tex storage.modifier.register.tex
+      captures:
+        1: punctuation.definition.backslash.tex
+      push: def-registername
+
+  def-registername:
+    - match: \{
+      scope: punctuation.definition.group.brace.begin.tex
+      push: register-name-group-body
+    - match: (\\){{cmdname}}
+      scope: entity.name.constant.tex
+      captures:
+        1: punctuation.definition.backslash.tex
+      pop: 1
+    - include: paragraph-pop
+    - include: else-pop
+
+  register-name-group-body:
+    - meta_scope: meta.group.brace.tex
+    - match: \}
+      scope: punctuation.definition.group.brace.end.tex
+      pop: 2
+    - match: (\\){{cmdname}}
+      scope: entity.name.constant.tex
+      captures:
+        1: punctuation.definition.backslash.tex
+
+
+###[ Dimensions ]###############################################################
+
+  tex-assignment:
+    - match: =
+      scope: keyword.operator.assignment.tex
+      pop: 1
+    - include: else-pop
+    - include: paragraph-pop
+
+
+  tex-dimension-value:
+    - match: (-?)(?:\d+(\.)\d*|(\.)\d+)
```

Comment:
Why? These are capture groups, which could possibly be empty. It's the style I've used everywhere when reworking number patterns in all syntaxes.

`??` is of no use/meaning here. It woudn't make a difference.

Reply:
This would be important if any other part of the pattern matching hyphons would be involved which should have less priority.

It wouldn't have any value here and is therefore not used in any other numbers pattern in any other syntax.